src/compiler/transformers/classFields.ts

```diff
@@ -1100,11 +1210,69 @@ namespace ts {
 
         function addPrivateIdentifierToEnvironment(node: PrivateClassElementDeclaration) {
             const text = getTextOfPropertyName(node.name) as string;
-            const { weakSetName } = getPrivateIdentifierEnvironment();
+            const env = getPrivateIdentifierEnvironment();
+            const { weakSetName } = env;

```

Comment:
nit: `const { weakSetName, classConstructor } = env;` would allow us to shorthand property names below.